code/modules/mob/living/carbon/human/species/station/prometheans.dm

```diff
@@ -150,15 +159,31 @@ var/datum/species/shapeshifter/promethean/prometheans
 			if (istype(T, /turf/simulated))
 				var/turf/simulated/S = T
 				S.dirt = 0
-			H.nutrition += rand(15, 45)
+			H.nutrition = min(400, max(0, rand(15, 30)))
 
 	// Heal remaining damage.
 	if(H.fire_stacks >= 0)
 		if(H.getBruteLoss() || H.getFireLoss() || H.getOxyLoss() || H.getToxLoss())
-			H.adjustBruteLoss(-heal_rate)
-			H.adjustFireLoss(-heal_rate)
-			H.adjustOxyLoss(-heal_rate)
-			H.adjustToxLoss(-heal_rate)
+			var/nutrition_cost = 0
+			var/nutrition_debt = H.getBruteLoss()
+			var/starve_mod = 1
+			if(H.nutrition <= 4)
+				starve_mod = 0.75
+			H.adjustBruteLoss(-heal_rate * starve_mod)
+			nutrition_cost = nutrition_debt - H.getBruteLoss()
```

Comment:
`nutrition_cost += nutrition_deb - H.get[damtype]Loss()` here and in the three instances below